__tests__/original.js

```diff
@@ -0,0 +1,22 @@
+"use strict"
+import produce, {original} from "../src/immer"
+
+describe("original", () => {
+    const baseState = {
+        a: [],
+        b: {}
+    }
+
+    it("should return the original from the proxy", () => {
+        const nextState = produce(baseState, draftState => {
+            expect(original(draftState)).toBe(baseState)
+            expect(original(draftState.a)).toBe(baseState.a)
+            expect(original(draftState.b)).toBe(baseState.b)
+        })
+    })
+
+    it("should return the same object from an object that is not proxied", () => {
```

Comment:
could you add a test that for a newline introduced object on the draft it returns `undefined`?

Reply:
And the same for a value that is a primitive

Reply:
I think returning `undefined` is slightly more intuitive / better supports fail-fast scenarios better. If you agree, I'll happily merge!